File: spoj/ACPC11B.cpp
#include <stdio.h>

int main(int argc, char const *argv[])
{
	int testCases;

	scanf("%d", &testCases);

	for (int i = 0 ; i < testCases ; i++)
	{
		int a;
		scanf("%d", &a);
		long long int arrA[a];
		for (int j = 0 ; j < a ; j++)
		{
			scanf("%lld", &arrA[j]); 
		}

		int b;
		scanf("%d", &b);
		long long int arrB[b];
		for (int j = 0 ; j < b ; j++)
		{
			scanf("%lld", &arrB[j]); 
		}

		long long int minimum = 10000005;

		for (int j = 0 ; j < a ; j++)
		{
			for (int k = 0 ; k < b ; k++)
			{
				long long int min = arrA[j] > arrB[k] ? (arrA[j] - arrB[k]) : (arrB[k] - arrA[j]);

				if (min < minimum)
				{
					minimum = min;
				}
			}
		}

		printf("%lld\n", minimum);
	}
	return 0;
}

File: spoj/ACPC11B.py
for i in range(0, int(raw_input())):
	arr_a = map(int, raw_input().split())
	arr_b = map(int, raw_input().split())

	min_ = 1000001
	for i in range(1, len(arr_a)):
		for j in range(1, len(arr_b)):
			temp_min = abs(arr_a[i] - arr_b[j])

			if temp_min < min_:
				min_ = temp_min

	print min_